server/src/main/java/uk/ac/cam/cl/echo/extrusionfinder/server/sourcer/crawlers/SeagateCrawler.java

```diff
@@ -0,0 +1,201 @@
+     * 1: 1.2 OD X 2.3 ID IN
+     * 2: 0.125 X 3.0 IN
+     * 3: 0.325 MM
+     */
+    private static final String F = MetadataParser.floatRegex();
+    private static final Pattern[] METADATA_FILTERS = new Pattern[] {
+            Pattern.compile("(("+ F +")( )DIA. (MM|IN)) (.*)"),
+            Pattern.compile("(("+ F +") OD [xX] ("+ F +") ID (MM|IN)) (.*)"),
+            Pattern.compile("(("+ F +") [xX] ("+ F +") (MM|IN)) (.*)"),
+            Pattern.compile("(("+ F +")( )(MM|IN)) (.*)")
+    };
+
+    /* This is the stream that we manipulate via side effects.
+     * Note that if used, it HAS to be assigned with configure method.
+     */
+    private static List<Part> parts;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void configure(List<Part> parts) {
+        this.parts = parts;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String[] getSeeds() {
+        return SEEDS;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getManufacturerId() {
+        return MANUFACTURER_ID;
+    }
+
+    /**
+     * Filters out the pages we want to visit.
+     * @param referringPage The parent page of this URL
+     * @param url           The current URL that we want to check
+     * @return              True if the URL should be visited
+     */
+    @Override
+    public boolean shouldVisit(Page referringPage, WebURL url) {
+
+        String href = url.getURL().toLowerCase();
+        return !FILTERS.matcher(href).matches() && (href.startsWith(SEEDS[0]));
+    }
+
+    /**
+     * Process the page, namely get information about parts.
+     * @param page  The fetched page
+     */
+    @Override
+    public void visit(Page page) {
+
+        String url = page.getWebURL().getURL();
+
+        if (page.getParseData() instanceof HtmlParseData) {
+            HtmlParseData htmlParseData = (HtmlParseData) page.getParseData();
+
+            String html = htmlParseData.getHtml();
+            MetadataParser hp = new MetadataParser(html, url);
+            extractData(hp);
+        }
+    }
+
+    /* Extracts metadata from the page. Add found parts to the internal list
+     * of parts.
+     */
+    private void extractData(MetadataParser hp) {
+
+        Elements productNodes = hp.getHtmlDoc().getElementsByClass("node");
+        for (Element productNode : productNodes) {
+
+            // get the product code
+            String productId = hp.selectSingleText(productNode, "div.product-name");
+            if (productId.isEmpty()) {
+                continue;
+            }
+            productId = productId.toUpperCase();
+
+            // get the link to the product, deal with seagate's link problems
+            String link = hp.getLink(productNode);
+            link = fixLink(link);
+
+            // get the image of the product
+            String image = hp.selectSingleAttr(productNode, "img[src]", "abs:src");
+            if (!image.isEmpty()) {
```

Comment:
What does it mean for the image to be empty? Can we keep processing this part if it is?